### PR TITLE
[may18] Readme updates: Landing page & UX

### PR DESCRIPTION
**Pull request by cre8ivepark**
Overview
---
**Same updates on Dev branch = #2029. Only folder names are different(HoloToolkit vs MixedRealityToolkit)**
Updating outdated readme page for UX. Separated readme for TextPrefab. Added links to the Spatial Mapping/Processing/Understanding example scenes on the landing page to improve the discoverability. Added thumbnail images for these example scenes.

Changes
---
- Updated [README.MD](https://github.com/cre8ivepark/MixedRealityToolkit-Unity/blob/Dev_Working_Branch_readme_images/README.md) on the landing page.
2018-05-01 14_13_02-cre8ivepark_mixedrealitytoolkit-unity at dev_working_branch_readme_images

- Updated [README.MD](https://github.com/cre8ivepark/MixedRealityToolkit-Unity/tree/Dev_Working_Branch_readme_images/Assets/MixedRealityToolkit/UX) in UX folder.
- Added new [README_TextPrefab.MD](https://github.com/cre8ivepark/MixedRealityToolkit-Unity/blob/Dev_Working_Branch_readme_images/Assets/MixedRealityToolkit/UX/README_TextPrefab.md) in UX folder

### CODE REVIEW

File: Assets/HoloToolkit/UX/README.md [hunk 1]
```diff
@@ -2,76 +2,34 @@
...
...[Dialog](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/Dev_Working_Branch/Assets/MixedRealityToolkit-Examples/UX/Readme/README_DialogsExample.md) / [Progress](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/Dev_Working_Branch/Assets/MixedRealityToolkit-Examples/UX/Readme/README_ProgressExample.md) | [![Solver system](/External/ReadMeImages/MRTK_SolverSystem.jpg)](/Assets/HoloToolkit-Examples/Utilities/Readme/README_SolverSystem.md) [Solver system](/Assets/HoloToolkit-Examples/Utilities/Readme/README_SolverSystem.md) | [![Interactive button demos](/External/ReadMeImages/MRTK_InteractiveButtons.jpg)](/Assets/HoloToolkit-Examples/UX/Readme/README_InteractiveButtonComponents.md) [Interactive button demos](/Assets/HoloToolkit-Examples/UX/Readme/README_InteractiveButtonComponents.md) | [![ToolTip & Label](/External/ReadMeImages/MRTK_Tooltip.jpg)](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/Dev_Working_Branch/Assets/MixedRealityToolkit-Examples/UX/Readme/README_TooltipExample.md) [ToolTip & Label](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/Dev_Working_Branch/Assets/MixedRealityToolkit-Examples/UX/Readme/README_TooltipExample.md) |
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/UX/README.md]: Same issue here too per https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/2029#discussion_r185561364

File: README.md [hunk 2]
```diff
@@ -63,6 +63,12 @@ To get up and going as quickly as possible, here are some guides to help you get
...g) [Dialog](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/Dev_Working_Branch/Assets/MixedRealityToolkit-Examples/UX/Readme/README_DialogsExample.md) / [Progress](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/Dev_Working_Branch/Assets/MixedRealityToolkit-Examples/UX/Readme/README_ProgressExample.md) | [![Solver system](External/ReadMeImages/MRTK_SolverSystem.jpg)](/Assets/HoloToolkit-Examples/Utilities/Readme/README_SolverSystem.md) [Solver system](/Assets/HoloToolkit-Examples/Utilities/Readme/README_SolverSystem.md) | [![Interactive button demos](External/ReadMeImages/MRTK_InteractiveButtons.jpg)](/Assets/HoloToolkit-Examples/UX/Readme/README_InteractiveButtonComponents.md) [Interactive button demos](/Assets/HoloToolkit-Examples/UX/Readme/README_InteractiveButtonComponents.md) | [![ToolTip & Label](External/ReadMeImages/MRTK_Tooltip.jpg)](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/Dev_Working_Branch/Assets/MixedRealityToolkit-Examples/UX/Readme/README_TooltipExample.md) [ToolTip & Label](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/Dev_Working_Branch/Assets/MixedRealityToolkit-Examples/UX/Readme/README_TooltipExample.md) |
```
Comment (StephenHodgson) [inline README.md]: These should be relative links too

File: Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Dialogs Example
+<img src="/External/ReadMeImages/MRTK_Dialog.jpg" width="650">
```
Comment (keveleigh) [inline Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md]: This image doesn't seem to exist.

File: Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+In summary, the steps to create and launch a Dialog are:
+In a LaunchFunction() called by a Coroutine:
+1. Load Dialog Prefab into memory using Resources.Load("Dialog").
+2. Create an instance of the Dialog Prefab using Dialog.Open(...). This also Opens and displays the Dialog.
+3. Attach OnClosed handler to the dialog.OnClosed event.
+
+## Properties
+The Dialog will stay in the user's field of view while the user can use gaze or controller targeting to make a button selection. Tag-along and billboarding interaction is achieved by using **Solver scripts**. When you select Dialog.prefab, you will be able to find Solver scripts for the size and distance control. In **SolverConstantViewSize**, you can adjust **Target View Percent V** value to modify the size of the Dialog window.
+
+You can find more detailed information about the [Solver System on this README page](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/Dev_Working_Branch/Assets/HoloToolkit-Examples/Utilities/Readme/README_SolverSystem.md).
```
Comment (StephenHodgson) [inline Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md]: relative link needed

File: README.md (hunk 2 above)
Comment (cre8ivepark) [reply README.md]: Updated

File: Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md (hunk 3 above)
Comment (cre8ivepark) [reply Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md]: Added missing images

File: Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# Dialogs Example
+<img src="/External/ReadMeImages/MRTK_Dialog.jpg" width="650">
+Dialog is a transient UI element which appears when something happens that requires notification, approval, or additional information from the user. Dialogs block interactions with the app window until being explicitly dismissed. They often request some kind of action from the user. Dialog provides one or two button options. It works on both HoloLens(gaze and gesture) and Immersive headset(motion controllers).
+
+
+![Dialogs on HoloLens and Immersive Headset](/External/ReadMeImages/MRTK_Dialog_Devices.jpg)
```
[the remark this answers is not in the document]
Comment (cre8ivepark) [reply Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md]: Added missing images

File: Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+1. Load Dialog Prefab into memory using Resources.Load("Dialog").
+2. Create an instance of the Dialog Prefab using Dialog.Open(...). This also Opens and displays the Dialog.
+3. Attach OnClosed handler to the dialog.OnClosed event.
+
+## Properties
+The Dialog will stay in the user's field of view while the user can use gaze or controller targeting to make a button selection. Tag-along and billboarding interaction is achieved by using **Solver scripts**. When you select Dialog.prefab, you will be able to find Solver scripts for the size and distance control. In **SolverConstantViewSize**, you can adjust **Target View Percent V** value to modify the size of the Dialog window.
+
+You can find more detailed information about the [Solver System on this README page](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/Dev_Working_Branch/Assets/HoloToolkit-Examples/Utilities/Readme/README_SolverSystem.md).
+
+<img src="/External/ReadMeImages/MRTK_Dialog_Inspector.jpg" width="450">
```
[the remark this answers is not in the document]
Comment (cre8ivepark) [reply Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md]: Added missing images

File: Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+## Dialog prefab and DialogShell script
+Dialog prefab is located under HoloToolkit/UX/Prefabs folder and dynamically instantiated with Unity's Instantiate() function. This prefab contains **DialogShell** script and **Solver** scripts.
+
+## How to display a Dialog control
+You can find the code example in **LaunchDialogScript.cs** under SceneContent > SquareButton. It demonstrates the instantiation and response to a Dialog using the function Launch Dialog. The Dialog can be specified to have a single button, useful for communicating information to the user. The Dialog can also be configured to display two buttons, allowing the user to make a choice. The name of the selected button is returned to the script that opens the Dialog. 
+
+An instance of the Dialog is created using: 
+ 
+//fill dialogPrefab in the inspector:
```
[the remark this answers is not in the document]
Comment (cre8ivepark) [reply Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md]: Thanks! applied same changes to Dev branch PR too.

File: Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md (hunk 4 above)
Comment (cre8ivepark) [reply Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md]: Updated. Dev branch PR as well.

File: Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+To close the ProgressIndicator, call the Close() function and wait for the IsLoading state to become false.
+<pre>
+LoadingDialog.Instance.Close();
+while (LoadingDialog.Instance.IsLoading)
+{
+    yield return null;
+}
+</pre>
+
+
```
[the remark this answers is not in the document]
Comment (cre8ivepark) [reply Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md]: Updated. Dev branch PR as well.

File: Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
+# Progress Example
+<img src="/External/ReadMeImages/MRTK_Progress1.jpg" width="650">
+A progress control provides feedback to the user that a long-running operation is underway. It can mean that the user cannot interact with the app when the progress indicator is visible, and can also indicate how long the wait time might be, depending on the indicator used.
+
+This example scene illustrates the use of the ProgressIndicator prefab to create a progress Indicator with several options for appearance.
+
+## Demo Video
+https://gfycat.com/JaggedDimLacewing
```
Comment (davidkline-ms) [inline Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md]: Who created this video? Do we own it?

File: Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+![Progress Types](/External/ReadMeImages/MRTK_Progress4.jpg)
+
+
+## How to display a Progress control
+In the example scene, each button is labeled with the variation of the ProgressIndicator that is demonstrated. There is an instance of the ProgressIndicator prefab in the scene which initializes and runs with arguments obtained from the example button that is clicked. 
+
+In these buttons, you can find **ProgressButton** script along with Compound Button scripts. **ProgressButton** script contains simple code to call LaunchProgress() in **ProgressExamples** script on OnButtonClicked event. In **ProgressExamples** script, you can find the code for displaying Progress control.
+
+It looks like this: 
+<pre>
```
Comment (davidkline-ms) [inline Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md]: Should this be formatted as code with the surrounding ``` markers?

File: Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+                            ProgressIndicator.IndicatorStyleEnum.None,
+                            ProgressIndicator.ProgressStyleEnum.None,
+                            ProgressIndicator.MessageStyleEnum.Visible,
+                            LeadInMessage);
+
+StartCoroutine(LoadOverTime(LoadTextMessage));
+</pre>
+
+The LoadOverTime function, also found in this script, fakes a loading sequence. To implement the ProgressIndicator during an actual loading sequence, Start a Coroutine with a function like LoadOverTime. In your LoadOverTime function, make sure to update the ProgressIndicator object with the actual loading progress.
+<pre>
```
Comment (davidkline-ms) [inline Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md]: Should this be formatted as code with the surrounding ``` markers?

File: Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+while (your loading operation is not complete)
+{
+    LoadingDialog.Instance.SetMessage( <loading progress message> );
+    LoadingDialog.Instance.SetProgress<percent loaded>
+    yield return new WaitForSeconds(<amount of time between updates ie 0.25 seconds>);
+}
+</pre>
+
+You can give an optional message to be displayed by the ProgressIndicator when load is complete, such as "finished!".
+<pre>
```
Comment (davidkline-ms) [inline Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md]: Should this be formatted as code with the surrounding ``` markers?

File: Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+ProgressIndicator.Instance.SetMessage(FinishMessage);
+ProgressIndicator.Instance.SetProgress(1f);
+yield return new WaitForSeconds(<amount of time to hold message ie 1.5 secs>);
+</pre>
+
+To close the ProgressIndicator, call the Close() function and wait for the IsLoading state to become false.
+<pre>
```
Comment (davidkline-ms) [inline Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md]: Should this be formatted as code with the surrounding ``` markers?

File: Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+Dialog is a transient UI element which appears when something happens that requires notification, approval, or additional information from the user. Dialogs block interactions with the app window until being explicitly dismissed. They often request some kind of action from the user. Dialog provides one or two button options. It works on both HoloLens(gaze and gesture) and Immersive headset(motion controllers).
+
+
+![Dialogs on HoloLens and Immersive Headset](/External/ReadMeImages/MRTK_Dialog_Devices.jpg)
+
+## Demo Video
+https://gfycat.com/ConventionalDirtyKiskadee
+
+## Dialog prefab and DialogShell script
+Dialog prefab is located under HoloToolkit/UX/Prefabs folder and dynamically instantiated with Unity's Instantiate() function. This prefab contains **DialogShell** script and **Solver** scripts.
```
Comment (davidkline-ms) [inline Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md]: Link to HoloToolkit/UX/Prefabs?

File: Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md [hunk 15]
```diff
@@ -0,0 +1,48 @@
+# Dialogs Example
+<img src="/External/ReadMeImages/MRTK_Dialog.jpg" width="650">
+Dialog is a transient UI element which appears when something happens that requires notification, approval, or additional information from the user. Dialogs block interactions with the app window until being explicitly dismissed. They often request some kind of action from the user. Dialog provides one or two button options. It works on both HoloLens(gaze and gesture) and Immersive headset(motion controllers).
+
+
+![Dialogs on HoloLens and Immersive Headset](/External/ReadMeImages/MRTK_Dialog_Devices.jpg)
+
+## Demo Video
+https://gfycat.com/ConventionalDirtyKiskadee
```
Comment (davidkline-ms) [inline Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md]: Who created this video?

File: Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+
+![Progress Type Buttons](/External/ReadMeImages/MRTK_Progress3.jpg)
+
+Different types of Progress control can be opened by pressing these buttons in the scene.
+
+
+
+## ProgressIndicator Prefab
+
+The **ProgressIndicator prefab** is located in HoloToolkit/UX/Prefabs/Progress. It is composed of **ProgressIndicator** script and **Solver** scripts for the billboarding and tag-along behavior.
```
Comment (davidkline-ms) [inline Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md]: Link to HoloToolkit/UX/Prefabs/Progress?

File: Assets/HoloToolkit-Examples/UX/Readme/README_TooltipExample.md [hunk 17]
```diff
@@ -0,0 +1,37 @@
+# Tooltip Example
+<img src="/External/ReadMeImages/MRTK_Tooltip.jpg">
+This example scene demonstrates an implementation of the ToolTip user interface element. Tooltips are usually used to convey a hint or extra information upon closer inspection of an object. ToolTip can be used to explain button inputs on the motion controllers or to label objects in the physical environment.
+
+## Demo Video
+https://gfycat.com/WarmOblongBilby
```
Comment (davidkline-ms) [inline Assets/HoloToolkit-Examples/UX/Readme/README_TooltipExample.md]: Who created this video?

Comment (cre8ivepark) [reply Assets/HoloToolkit-Examples/UX/Readme/README_TooltipExample.md]: @davidkline-ms I did.

File: Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md (hunk 9 above)
Comment (cre8ivepark) [reply Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md]: @davidkline-ms Yes, I created it.

File: Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md (hunk 10 above)
Comment (cre8ivepark) [reply Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md]: Great, I'll replace <pre> tags with ```

Comment (cre8ivepark) [reply Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md]: Replaced all < pre > tags

File: Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md (hunk 14 above)
Comment (cre8ivepark) [reply Assets/HoloToolkit-Examples/UX/Readme/README_DialogsExample.md]: Added links

File: Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md (hunk 16 above)
Comment (cre8ivepark) [reply Assets/HoloToolkit-Examples/UX/Readme/README_ProgressExample.md]: Added links

**Comment by cre8ivepark** [comment]
Applied same changes to #2029 for Dev branch.

**Pull request merged**